keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,696 @@
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable, can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: Editing the existing component config file to add the new minimum required field (metadata.name) and restarting the component.
+    - Will enabling / disabling the feature require downtime of the control
+      plane? 
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+
+ No, these new fields don't control component behavior. There may be some impact to existing tests making use of previously overloaded component configs with fields of the same name if existing values are out of bounds.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g. if this is runtime
+  feature, can it break the existing applications?).
+
+Yes, this can be disabled once it has been enabled. The existing component API's have a lenient path implementation for the strict decoders that allow for previously overloaded configs with fields of the same name to pass validation.
```

Comment:
We can probably say: "Users can choose whether to set the field or not. As far as a component is concerned, it is safe to unset the field after it has been set. If the user had already set the field, it will continue to work after this change. If the user chooses to set the field in order to depend on tooling that requires it, then unsetting the field would require them to return to their previous approach. Whether the schema can be removed in the future depends on the stability level of each API we add it to."

I don't want to rely too much on "it's currently non-strict" as the "it's safe" measure, since we could move to strict in the future (though that transition will probably happen across componentconfig version boundaries anyway).